BackEnd/src/main/java/com/unosquare/admin_core/back_end/repository/EventRepository.java

```diff
@@ -13,11 +15,44 @@
 
     List<Event> findByEmployee(Employee employee);
 
-    Event findByEmployeeAndStartDateAndEndDateAndEventType(Employee employee, LocalDate startDate, LocalDate endDate, EventType eventType);
+    Event findByEmployeeAndStartDateAndEndDateAndEventType(Employee employee, LocalDate startDate, LocalDate endDate,
+                                                           EventType eventType);
 
-    List<Event> findByStartDateBetweenAndEventType(LocalDate rangeStart, LocalDate rangeEnd,EventType eventType);
+    List<Event> findByStartDateBetweenAndEventType(LocalDate rangeStart, LocalDate rangeEnd, EventType eventType);
 
     List<Event> findByEventStatusAndEventType(EventStatus eventStatus, EventType eventType);
 
     List<Event> findByEventType(EventType eventType);
+
+    @Query(value = "SELECT COUNT(e.eventType) " +
+            "FROM Event e " +
+            "WHERE " +
+            "e.employee.employeeId = :employeeId " +
+            "AND " +
+            "(e.startDate BETWEEN :startDate AND :endDate) OR " +
+            "(e.endDate BETWEEN :startDate AND :endDate) OR " +
+            "(e.startDate <= :startDate AND e.endDate >= :endDate) ")
+    Integer getCountOfNumberOfExistingEventsByEmployee(@Param("employeeId") Employee employee,
+                                                    @Param("startDate") LocalDate startDate,
+                                                    @Param("endDate") LocalDate endDate);
+
+//    Sub query to pull back all events in a calender year by employee
+//    Sum to add up all events within a calender year to be subtracted from totalHoliday count
+//    @Query(value =
+//            "SELECT SUM (DATE_PART('DAY', totalEvents.end_date::date) - DATE_PART('DAY', totalEvents.start_date::date))" +
+//                    "FROM " +
+//                    "(SELECT * " +
+//                    "FROM Event " +
+//                    "WHERE " +
+//                    "employee_id = :employeeId " +
+//                    " AND " +
+//                    "(start_date >= '2018-01-01') AND " +
+//                    "(end_date <= '2018-12-31') " +
+//                    ")totalEvents", nativeQuery = true)
+//    Long getCountOfTotalEventsInYearMadeByEmployee(@Param("employeeId") Employee employee);
+
+    @Query(value = "SELECT e.employee.employeeId " +
```

Comment:
Do you need just need an employeeId at this point ? What is it being used for? If you are then using that Id to pull back an employee it would be more efficient to just return the full employee entity? Again i'm not sure if this is the use case?

Reply:
I'm not sure which query you're referencing here. But if it's for the 'Admin/Team leads can not approve their own holidays', having back just the 'employeeID' that's on the specific event/holiday would surely be enough. Rather than pull back a full employee entity. If we are using the 'employeeCredentialsViewModel.getUserId()' on the holiday controller then comparing the two ID's is simply all that's needed.